lib/request.ts

```diff
@@ -321,7 +323,14 @@ export const request = (
           bufferLength
         );
         hasRequestEnded = true;
-        const body = Buffer.concat(chunks, bufferLength).toString("utf8");
+
+        let body;
+        const bufferedBody: Buffer = Buffer.concat(chunks, bufferLength);
```

Comment:
I have noticed the type of body in the `ServiceClientResponse` is any, maybe we could type it as `Buffer | string | object`.

Bonus points if we can type it conditionally based on the options. :smile:

Reply:
I think it would be more convenient for clients if we decode based on the `content-type` in the response